forms-flow-web/src/components/Form/Item/View.js

```diff
@@ -415,7 +417,7 @@ const View = React.memo((props) => {
               }}
               onCustomEvent={(evt) => onCustomEvent(evt, redirectUrl)}
             />
-          ) : formStatus === "inactive" || !formStatus ? (
+          ) : !processLoadError && (formStatus === "inactive" || !formStatus)  ? (
```

Comment:
we can add like this right?
`if(processLoadError){
return <NotFound/> }`

as first return

Reply:
The page and form data are loading separately, if we edit the code as  mentioned above, the top portion of the page will be displayed initially even for invalid id. And as it is invalid id, the entire page will then  lead to 404.

Reply:
As @shuhaib-aot  mentioned we can make the Ui rendering less complex but it would be better to write a function and check the conditions alltogether and return the components. The code conditions become more complex to understand with multiple ternary inside conditions and will be difficult to maintain future code